docs/safe_lambda_deployments.rst

```diff
@@ -180,20 +189,39 @@ Hooks are extremely powerful because:
   you can directly invoke the version to run integration tests or even pre-warm the Lambda containers before exposing
   to production traffic.
 
+    NOTE: The event payload delivered to the Hook function will not contain the Lambda ARN to be tested.
+    We recommend adding an Envrionment variable to the Hook function that maintains the current version of the function requiring safe deployments
+
+.. code:: yaml
+
+  CurrentVersion: !Ref MySafeLambdaFunction.Version
+
+
 - **Hooks are executed per-function**: Each Lambda function gets its own PreTraffic and PostTraffic hook (technically
   speaking hooks are executed once per DeploymentGroup, but in this case the DeploymentGroup contains only one Lambda
   Function). So you can customize the hooks logic to the function that is being deployed.
 
-Checkout the examples folder for an example for how to create a hook function.
-
     NOTE: If the Hook functions are created by the same SAM template that is deployed, then make sure to turn off
-    traffic shifting deployments for the hook functions. 
+    traffic shifting deployments for the hook functions. Also, you should not rely on SAM to create the default function Role, since it
```

Comment:
This is very misleading. Best practice is to let SAM create the default Role. It is true that the Generated role will not have the 'codedeploy:PutLifecycleEventHookExecutionStatus' but this should be added through the Policies block. We should not be suggesting to complete manage a Role for a function. There are very few use-cases to have customers create the Role and attach it to the function. Please update this section accordingly to use Policies blocks. I would rather just leave out the Role completely.

Reply:
Done. Policies added.

Reply:
"Also, you should not rely on SAM to create the default function Role, since it will not contain the necessary permissions to call the CodeDepoloy APIs or Invoke your new Lambda function for testing."

They should rely on the Role SAM creates. I understand what you are trying to get across here but I think it can be clearer. How about something like:
The Role SAM generates for a Lambda Execution Role does not include all permissions needed for Per and Post hook functions. To achieve this, use the Policies attribute ....

Reply:
Agreed - updated.